src/openrct2/drawing/Text.cpp

```diff
@@ -164,9 +164,13 @@ void gfx_draw_string_centred(
 
 void gfx_draw_string_right(rct_drawpixelinfo* dpi, rct_string_id format, void* args, uint8_t colour, int32_t x, int32_t y)
 {
-    DrawTextCompat(dpi, x, y, format, args, colour, TextAlignment::RIGHT);
+    gfx_draw_string_right(dpi, format, args, colour, { x, y });
 }
 
+void gfx_draw_string_right(rct_drawpixelinfo* dpi, rct_string_id format, void* args, uint8_t colour, const ScreenCoordsXY& coords)
```

Comment:
```suggestion
void gfx_draw_string_right(
    rct_drawpixelinfo* dpi, rct_string_id format, void* args, uint8_t colour, const ScreenCoordsXY& coords)
```
The only thing missing is that the format is broken, you need to apply this change.